### PR TITLE
Add 4 tests

### PR DESCRIPTION
**Pull request by gRoussac**

### CODE REVIEW

**Review by cryofracture — APPROVED** [review]

File: tests/src/integration_tests.rs [hunk 1]
```diff
@@ -1,107 +1,137 @@
...
+            None,
+            ENTRY_POINT_INVENTORY_GET,
+            runtime_args! {
+                RUNTIME_KEY_NAME => item
+            },
+        )
+        .build();
+
+        // deploy the contract.
+        let ret = builder.exec(execute_request).commit().get_exec_result(1);
```
Comment (gRoussac) [inline tests/src/integration_tests.rs:119]: @cryofracture probably this line is wrong, But I don't know how it is supposed to work here precisely

File: contract/src/main.rs [hunk 2]
```diff
@@ -180,20 +187,18 @@ pub extern "C" fn call() {
...
     );
 
     // Store the contract version in the context's named keys
     let version_uref = storage::new_uref(contract_version);
     runtime::put_key(CONTRACT_VERSION_KEY, version_uref.into());
 
     // Create a named key for the contract hash
-    runtime::put_key(DICT_NAME, stored_contract_hash.into());
-
+    runtime::put_key(CONTRACT_HASH, stored_contract_hash.into());
```
Comment (gRoussac) [inline contract/src/main.rs:199]: [⚠️](https://emojipedia.org/warning/) DICT_NAME <> CONTRACT_HASH

File: contract/src/main.rs [hunk 3]
```diff
@@ -227,18 +232,20 @@ pub extern "C" fn call() {
...
 // }
 //
+
+// It does not make sense on gas point of view to query a dict with a payable entrypoint
+// Dictionnaries are in state, no need to query them with an entry point
+// Its uref is stored in named keys of admin and contract named keys thus values are accessible directly through dictionnarie queries
 #[no_mangle]
 pub extern "C" fn inventory_get() {
     let dictionary_item_key: String = runtime::get_named_arg(RUNTIME_KEY_NAME);
-    let uref: URef = runtime::get_key(RUNTIME_SEED_REF)
```
Comment (gRoussac) [inline contract/src/main.rs:233]: [⚠️](https://emojipedia.org/warning/) Watch out how many dicts you manipulate

File: contract/src/main.rs [hunk 4]
```diff
@@ -88,90 +88,97 @@ impl From<Error> for ApiError {
...
-//     store_entry_points.add_entry_point(EntryPoint::new(
-//         ENTRY_POINT_INVENTORY_DEC,
-//         Vec::new(),
-//         CLType::Unit,
-//         EntryPointAccess::Public,
-//         EntryPointType::Contract,
-//     ));
-
-  // store dict seed URef
-    let dict_seed = storage::new_uref(dict_seed_ref);
```
Comment (gRoussac) [inline contract/src/main.rs:170]: [⚠️](https://emojipedia.org/warning/) dict_seed_ref is already a uref. Nesting a uref into a uref does not seem necessary and it is pretty complex to get why the key is missing when actually you query later the outer uref and not the inner uref.

File: contract/src/main.rs [hunk 5]
```diff
@@ -88,90 +88,97 @@ impl From<Error> for ApiError {
...
+    //         }
+    //     }
 
     // Create entry points for this contract
     let mut store_entry_points = EntryPoints::new();
 
     store_entry_points.add_entry_point(EntryPoint::new(
         ENTRY_POINT_INVENTORY_GET,
-        Vec::new(),
+        vec![Parameter::new(RUNTIME_KEY_NAME, CLType::String)],
```
Comment (gRoussac) [inline contract/src/main.rs:154]: watch out not to forget arguments in this list when creating an entry point

**Pull request merged**